controllers/metal3.io/host_state_machine_test.go

```diff
@@ -649,6 +926,14 @@ func (m *mockProvisioner) Delete() (result provisioner.Result, err error) {
 	return m.getNextResultByMethod("Delete"), err
 }
 
+func (m *mockProvisioner) Detach() (result provisioner.Result, err error) {
+	res := m.getNextResultByMethod("Detach")
+	if res.ErrorMessage == "" {
+		m.isDetached = true
```

Comment:
That looks rather unusual, and as previously suggested I think it could be removed.
In case it would be necessary to verify that a provisioner method has been invoked, the mock could have a generic mechanism to record the calls (personally not a big fan of such approach, but sometimes it could be required)

Reply:
Currently we just assume the mockProvisioner methods have been called without checking, my goal here was to prove the Detach() had actually been called, but given the scope of host_state_machine_test.go is the HSM primarily I can remove it.

Reply:
I replaced this by a generic `calledNoError` method on the mockProvisioner - personally I think there's value in proving the Provisioner method got called (and also that it didn't in all the states where we don't expect `Detach()` to happen)